### PR TITLE
ci: refactoring from adding material tests to master ci (PATCH)

### PR DESCRIPTION
**Pull request by jelbourn**
This cherry-picks changes from #28197 to minimize the difference in
the ci infrastructure between master the the current patch branch

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -481,7 +506,8 @@ workflows:
       - lint
       - test
       - test_ivy_aot
-      - build-packages-dist
+      - build-npm-packages
+      - build-ivy-npm-packages
```
Comment (IgorMinar) [inline .circleci/config.yml]:
```suggestion
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -527,9 +553,10 @@ workflows:
             # Only publish if `aio`/`docs` tests using the locally built Angular packages pass
             - test_aio_local
             - test_docs_examples
-            # Get the artifacts to publish from the build-packages-dist job
+            # Get the artifacts to publish from the build-npm-packages job
             # since the publishing script expects the legacy outputs layout.
-            - build-packages-dist
+            - build-npm-packages
+            - build-ivy-npm-packages
```
Comment (IgorMinar) [inline .circleci/config.yml]:
```suggestion
```

**Review by IgorMinar — APPROVED** [review]
The PR fails on CI due to some silliness.

I also proposed some small changes to drop the ivy packages job because it would only waste resources on CI.

otherwise lgtm. thanks!

**Comment by matsko** [comment]
Landed as https://github.com/angular/angular/commit/5b72d4d67629887d5949383ccab58518eecf0369

**Pull request closed without merging**